File: integration-tests/mongodb-panache/src/test/java/io/quarkus/it/mongodb/panache/model/BikeV.java
package io.quarkus.it.mongodb.panache.model;

import java.util.Objects;

import org.bson.types.ObjectId;

import io.quarkus.mongodb.panache.PanacheMongoEntity;
import io.quarkus.mongodb.panache.common.MongoEntity;
import io.quarkus.mongodb.panache.common.Version;

@MongoEntity
public class BikeV extends PanacheMongoEntity {

    @Version
    public Long version;

    public BikeV() {
    }

    public BikeV(ObjectId id, Long version) {
        this.version = version;
        this.id = id;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o)
            return true;
        if (!(o instanceof BikeV))
            return false;
        BikeV bikeV = (BikeV) o;

        return Objects.equals(version, bikeV.version)
                && Objects.equals(id, bikeV.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(version, id);
    }

    @Override
    public String toString() {
        return "BikeV{" +
                "version=" + version +
                ", id=" + id +
                '}';
    }
}

File: integration-tests/mongodb-panache-kotlin/src/test/kotlin/io/quarkus/it/mongodb/panache/model/BikeV.kt
package io.quarkus.it.mongodb.panache.model

import io.quarkus.mongodb.panache.common.MongoEntity
import io.quarkus.mongodb.panache.common.Version
import io.quarkus.mongodb.panache.kotlin.PanacheMongoEntity
import org.bson.codecs.pojo.annotations.BsonCreator
import org.bson.codecs.pojo.annotations.BsonProperty
import org.bson.types.ObjectId

@MongoEntity
class BikeV @BsonCreator constructor(
    @BsonProperty("modelYear") var modelYear: Int?,
    @BsonProperty("version") @Version
    var version: Long?
) : PanacheMongoEntity() {

    constructor(id: ObjectId?, modelYear: Int?, version: Long?) : this(modelYear, version) {
        this.id = id
    }
}
